app/app.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python3
+"""
+Created on Mon Dec 13 11:05:11 2021
+
+1. Runs the Mobotix C++ sampler with the given arguments.
+2. The .rgb files are too large, so we used ffmpeg to convert to JPG.
+"""
+
+import argparse
+import glob
+import logging
+import os
+import re
+import shutil
+import subprocess
+from select import select
+
+import timeout_decorator
+from waggle.plugin import Plugin
+
+# camera image fetch timeout (seconds)
+DEFAULT_CAMERA_TIMEOUT = 120
+
+
+def convertRGBtoJPG():
+    rgb_files = glob.glob("*.rgb")
+    for f_rgb in rgb_files:
+        # make JPEG file name
+        f_jpg = f_rgb.replace(".rgb", ".jpg")
+
+        # get the Width and Height of the image from filename
+        match_str = re.search("\d+x\d+", f_rgb)
+        image_dims = match_str.group()
+        subprocess.run(
+            [
+                "ffmpeg",
+                "-f",
+                "rawvideo",
+                "-pixel_format",
+                "bgra",
+                "-video_size",
+                image_dims,
+                "-i",
+                f_rgb,
+                f_jpg,
+            ]
+        )
+
+        logging.debug("Removing " + f_rgb)
+        os.remove(f_rgb)
+
+
+def renameFiles():
+    files = sum([glob.glob(f) for f in ("*.raw", "*.jpg", "*.csv")], [])
+    timestamp = []
+    new_filenames = []
+    for f in files:
+        timestamp_str, filename = f.split("_", 1)
+
+        timestamp.append(int(timestamp_str))
+        new_filenames.append(filename)
+
+        logging.debug("Renaming " + f)
+        os.rename(f, filename)
+
+    return new_filenames, timestamp
+
+
+def cleanup():
+    all_files = glob.glob("*")
+    for f in all_files:
+        logging.debug("Removing " + f)
+        try:
+            os.remove(f)
+        except IsADirectoryError:
+            shutil.rmtree(f)
+            pass
+
+
+@timeout_decorator.timeout(DEFAULT_CAMERA_TIMEOUT)
+def get_camera_frames(args):
+    cmd = [
+        "/thermal-raw",
+        "--url",
+        args.ip,
+        "--user",
+        args.user,
+        "--password",
+        args.password,
+        "--dir",
+        args.workdir,
+    ]
+    with subprocess.Popen(cmd, stdout=subprocess.PIPE) as process:
+        while True:
+            pollresults = select([process.stdout], [], [], 5)[0]
+            if pollresults:
+                output = pollresults[0].readline()
+                if output:
+                    m = re.search("frame\s#(\d+)", output.strip().decode())
+                    logging.info(output.strip().decode())
+                    if m and int(m.groups()[0]) > args.frames:
+                        logging.info("Max frame count reached, closing camera capture")
+                        return
+
+
+def main(args):
+
+    # create output directory if it does not exist and change WD.
+    if not os.path.exists(args.workdir):
+        os.makedirs(args.workdir)
+
+    os.chdir(args.workdir)
```

Comment:
do *not* change directories in your python code.  It is bad practice as it results in weird behavior when files are accessed.

Instead when searching for files, just search for all files in that directory.

```
>>> import glob
>>> glob.glob("data/*.*")
['data/a2.raw', 'data/a1.csv', 'data/a.rgb']
```